### PR TITLE
[ios] Zip file when attaching if it is secretly a directory

### LINKED ISSUE #5209: Attaching a directory from the mail editor crashes the app when sending
Normally it is not possible to attach directories, but for example exports from the GarageBand app have the extension .band, but are actually directories. Since they are displayed as normal files, it is possible to select them. When sending the mail, the app crashes because FileHandle expects files, not directories.

**To Reproduce**
1. open new mail editor
2. Click on attachment
3. click browse
4. Select a .band file
5. send the email
6. app crashes

## Test Notes
- [ ] Attaching files with the share function works as expected
- [ ] Attaching directories with the share function works as expected 
- [ ] Attaching files through the attach button when drafting a mail works as expected
- [ ] Attaching directory-files through the attach button works as expected.

### PR DESCRIPTION
**Pull request by wrdhub**
Throw error instead of crashing when not able to handle file

close #5209

### CODE REVIEW

File: app-ios/tutanota/Sources/Files/BlobUtil.swift [hunk 1]
```diff
@@ -39,10 +39,14 @@ class BlobUtil {
 
   
   func splitFile(fileUri: String, maxBlobSize: Int) async throws -> [String] {
-    let fileHandle = FileHandle(forReadingAtPath: fileUri)!
+    let fileHandle = FileHandle(forReadingAtPath: fileUri)
```
Comment (charlag) [inline app-ios/tutanota/Sources/Files/BlobUtil.swift:42]: I would use `guard let fileHandle = ...` instead but it's not that important

File: app-ios/tutanota/Sources/Files/IosFileFacade.swift [hunk 2]
```diff
@@ -37,7 +37,23 @@ class IosFileFacade : FileFacade {
...
-    return try await self.chooser.open(withAnchorRect: anchor)
+    let files = try await self.chooser.open(withAnchorRect: anchor)
+    var returnfiles = [String]()
+    for file in files {
+      let fileUrl = URL(fileURLWithPath: file)
+      let isDirectory: Bool = (try? fileUrl.resourceValues(forKeys: [.isDirectoryKey]))?.isDirectory == true
+      // This should only be for files, but sometimes a directory masquerading as a file can slip through (such as a .band file).
+      // In those cases we just zip and add it.
+      if isDirectory {
+        guard let destinationPath = try? zipFile(fileUrl: fileUrl) else {
```
Comment (charlag) [inline app-ios/tutanota/Sources/Files/IosFileFacade.swift:48]: I don't think we should quietly swallow these errors, we should rethrow it (with just `try`)

File: app-ios/tutanota/Sources/Files/IosFileFacade.swift [hunk 3]
```diff
@@ -141,10 +157,32 @@ class IosFileFacade : FileFacade {
     try data.write(to: URL(fileURLWithPath: filePath), options: .atomicWrite)
     return filePath
   }
+  
   func hashFile(_ fileUri: String) async throws -> String {
     return try await BlobUtil().hashFile(fileUri: fileUri)
   }
   
+  func zipFile(fileUrl: URL) throws -> String {
```
Comment (charlag) [inline app-ios/tutanota/Sources/Files/IosFileFacade.swift:165]: it's not necessarily a file I guess?

**Review by charlag — CHANGES_REQUESTED** [review]
Commit message is outdated now I think

**Comment by wrdhub** [comment]
> Commit message is outdated now I think

No, it still throws an error instead of crashing, I just decided to also handle zipping.

**Review by charlag — APPROVED** [review]

**Pull request merged**